wp-cli/vip-go-convert-to-utf8mb4.php

```diff
@@ -0,0 +1,348 @@
+		WP_CLI::line( '* dry run: ' . ( $this->dry_run ? 'yes' : 'no' ) );
+		WP_CLI::line( '* protecting utf8 masquerading as latin1: ' . ( $this->protect_masquerading_utf8 ? 'yes' : 'no' ) );
+		WP_CLI::line( '' );
+
+		// Validate starting charset to avoid catastrophe
+		WP_CLI::line( 'PREFLIGHT CHECKS' );
+		if ( in_array( $wpdb->charset, array( 'latin1', 'utf8', ), true ) ) {
+			WP_CLI::line( "* Expected charset (`{$wpdb->charset}`) found." );
+		} elseif ( 'utf8mb4' === $wpdb->charset ) {
+			WP_CLI::error( 'Site is already using `utf8mb4`. Aborting!' );
+			return;
+		} else {
+			WP_CLI::error( "Unacceptable starting encoding: `{$wpdb->charset}`. Aborting!" );
+			return;
+		}
+
+		// Describe scope
+		if ( is_multisite() ) {
+			WP_CLI::line( '* Multisite detected, so this process will convert all network and global tables, along with the blog tables for all sites.' );
+		} else {
+			WP_CLI::line( '* Single site detected, so global and blog-specific tables will be converted. Any multisite tables will be skipped.' );
+		}
+
+		// Describe tables to be converted
+		$this->get_tables();
+		$tables_count = number_format( count( $this->tables ) );
+		$tables_string = implode( ', ', $this->tables );
+
+		WP_CLI::line( "* Found {$tables_count} tables to check and potentially convert: {$tables_string}." );
+		WP_CLI::line( '' );
+
+		// Provide an opportunity to abort
+		WP_CLI::confirm( "Proceed with " . ( $this->dry_run ? 'DRY' : 'LIVE' ) . " RUN and " . ( $this->dry_run ? 'test converting' : 'potentially convert' ) . " {$tables_count} tables from `{$wpdb->charset}` to `utf8mb4`?" );
+		if ( ! $this->dry_run ) {
+			WP_CLI::confirm( 'ARE YOU REALLY SURE?' );
+		}
+		WP_CLI::line( '' );
+		WP_CLI::line( 'Proceeding...' );
+		WP_CLI::line( '' );
+
+		unset( $tables_count, $tables_string );
+
+		// Do the work we came here for
+		foreach ( $this->tables as $table ) {
+			WP_CLI::line( "Converting {$table}..." );
+
+			$converted = $this->maybe_convert_table_to_utf8mb4( $table );
+
+			if ( true === $converted ) {
+				WP_CLI::success( "Done with {$table}." );
+			} elseif ( false === $converted ) {
+				if ( $this->dry_run ) {
+					WP_CLI::line( "Table {$table} not converted during dry run." );
+				} else {
+					WP_CLI::warning( "Table {$table} not converted because it doesn't exist or doesn't contain convertible columns." );
+				}
+			} else {
+				WP_CLI::warning( 'Unknown response: ' . var_export( $converted, true ) );
+			}
+
+			WP_CLI::line( '' );
+		}
+
+		// Update DB's default charset/collation
+		$convert_db = $wpdb->query( 'ALTER DATABASE ' . DB_NAME . ' CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci' );
+		if ( $convert_db ) {
+			WP_CLI::success( "Set database to utf8mb4" );
+		} else {
+			WP_CLI::warning( "Could not update database default charset" );
+		}
+
+		// Wrap up
+		WP_CLI::line( '' );
+		WP_CLI::line( '' );
+		WP_CLI::line( 'DONE!' );
+		WP_CLI::line( 'Time to update sitemeta and reload web configs.' );
```

Comment:
Minor: would be nice to explicitly note what the meta key and value should be here.